activemodel/CHANGELOG.md

```diff
@@ -1,3 +1,19 @@
+*   Generates a method attribute_salt that returns the salt used to hash the password
+    generated by `has_secure_password`, it can be used for tracking changes
+    and as the source for `generates_token_for`.
+
+    ```ruby
+    class User < ActiveRecord::Base
+      has_secure_password
+
+      generates_token_for :password_reset, expires_in: 15.minutes do
+        password_salt.last(10)
```

Comment:
```suggestion
        password_salt&.last(10)
```

Reply:
In this case, the previous implementation would also fail, since password_digest is blank.

```ruby
  generates_token_for :password_reset, expires_in: 15.minutes do
    BCrypt::Password.new(password_digest).salt[-10..]
  end
  
  # -> invalid hash (BCrypt::Errors::InvalidHash)
```

Reply:
I think failing is okay, but `password_salt.last(10)` would fail with a `NoMethodError`, which isn't very helpful.  Also, people reading this example code may not even realize that `password_salt` can return `nil`, so `&.` offers a hint.